lib/CdeContext.tsx

```diff
@@ -1,83 +1,159 @@
 import {createContext, PropsWithChildren, useContext, useState} from 'react';
-import {Config, DatasetCDEMapping, InitParams, InputMapping, STEPS} from "./models.ts";
+import {
+    CDE,
+    Collection,
+    Config,
+    CustomDictionaryField,
+    DatasetMapping,
+    DatasetSample,
+    InitParams,
+    STEPS
+} from "./models.ts";
 import theme from "./theme/index.tsx";
 import {ThemeProvider} from "@mui/material";
 import CssBaseline from '@mui/material/CssBaseline';
+import {
+    validateDatasetMapping,
+    validateDatasetSample
+} from "./services/validatorsService.ts";
+import {mapDatasetSample, mapStringTableToDatasetMapping} from "./services/initialMappingService.ts";
+import {updateDatasetMappingRow} from "./services/updateMappingService.ts";
 
 export const CdeContext = createContext<{
-    labName: string;
-    config: Config;
-    inputMappings: InputMapping[][];
-    mapping: DatasetCDEMapping;
-    setMapping: (mapping: DatasetCDEMapping) => void;
+
+    name: string;
+    datasetSample: DatasetSample;
+    datasetMapping: DatasetMapping;
+    datasetMappingHeader: string[];
+    collections: Collection[]
+    config: Config
+
+
+    // UI Module
     step: number;
     setStep: (step: number) => void;
     loadingMessage: string | null;
     setLoadingMessage: (loadingMessage: string | null) => void;
     errorMessage: string | null;
     setErrorMessage: (errorMessage: string | null) => void;
-    isOpen: boolean;
     handleClose: () => void;
-    infoOpen: boolean;
```

Comment:
Removes isOpen,  infoOpen and setInfoOpen from the context since they can be managed by the CDE Modal component. cc @vidhya-metacell